### PR TITLE
Cleanup groups implementation

### PR DESCRIPTION
**Pull request by nmklotas**
Fix naming and formatting
Remove object properties in requests
Remove not needed ctors
Don't build queries for default parameters
Add QueryBuilders tests
Add GroupClient tests
Add more doc strings
Add Groups API to GitlabApiClient

### CODE REVIEW

File: src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs [hunk 1]
```diff
@@ -43,18 +46,20 @@ private static string GetVisibilityQueryValue(GroupsVisibility visibility)
                 case GroupsVisibility.Private:
                     return "private";
                 default:
-                    return "public";
```
Comment (ilijamitkov) [inline src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs]: if we are removing public as default we will need to add Public condition

File: src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs [hunk 2]
```diff
@@ -75,7 +80,7 @@ private static string GetOrderQueryValue(GroupsProjectsOrder order)
                 case GroupsProjectsOrder.LastiActivityAt:
                     return "last_activity_at";
                 default:
-                    return "created_at";
```
Comment (ilijamitkov) [inline src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs]: same here we need CreatedAt condition

File: src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs (hunk 1 above)
Comment (nmklotas) [reply src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs]: public condition is already added in the switch statement.

File: src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs (hunk 2 above)
Comment (nmklotas) [reply src/GitLabApiClient/Internal/Queries/GroupsProjectsQueryBuilder.cs]: createdAt also already added in  the switch statement

**Pull request merged**